modules/undertoneBidAdapter.js

```diff
@@ -0,0 +1,75 @@
+/**
+ * Adapter to send bids to Undertone
+ */
+
+import * as utils from 'src/utils';
+import { registerBidder } from 'src/adapters/bidderFactory';
+
+const BIDDER_CODE = 'undertone';
+const URL = '//hb.undertone.com/hb'; // //ads.undertone.com/hb
+
+export const spec = {
+  code: BIDDER_CODE,
+  isBidRequestValid: function(bid) {
+    if (bid && bid.params && bid.params.publisherId && bid.params.placementId) {
+      bid.params.publisherId = parseInt(bid.params.publisherId);
+      return true;
+    }
+  },
+  buildRequests: function(validBidRequests) {
+    const payload = {
+      'x-ut-hb-params': []
+    };
+    const timeout = window.PREBID_TIMEOUT || null;
```

Comment:
Information for prebid can't be accessed through the window object directly.  

As an alternative, you could pass the PREBID_TIMEOUT variable from the page code as another param value like the following:
```
params: {
              placementId: '10433394',
              publisherId: 12345,
              timeout: PREBID_TIMEOUT
              }
```

With this data, you could then access the value within your adapter by reading from the params and assign it to the bid.

Reply:
Our API enable us to pass the timeout value to the server, so we can response with nobid if we are not able to respond within this time frame. This should improve the user experience and reduce load from our server.
Is there another way to get the timeout you're using? Or maybe there is a constant value you're using? If so, we can just add it here.